modules/rubiconBidAdapter.js

```diff
@@ -451,9 +456,38 @@ export const spec = {
       data['coppa'] = 1;
     }
 
+    // if SupplyChain is supplied and contains all required fields
+    if (bidRequest.schain && hasValidSupplyChainParams(bidRequest.schain)) {
+      data.rp_schain = spec.serializeSupplyChain(bidRequest.schain);
+    }
+
     return data;
   },
 
+  /**
+   * Serializes schain params according to OpenRTB requirements
+   * @param {Object} supplyChain
+   * @returns {String}
+   */
+  serializeSupplyChain: function (supplyChain) {
+    const supplyChainIsValid = hasValidSupplyChainParams(supplyChain);
```

Comment:
This check occurs on line 460 as well as inside this function.

Is there a specific reason? Think if it is not valid we just do not set the `rp_schain` 

So seems like this:
```if (!supplyChainIsValid) return '';```
will never actually be hit right?

Reply:
The idea was that serializeSupplyChain() could be used directly from other methods if need be, and to eventually extract that method to a utils lib for other modules to use. I can remove it though.

Reply:
I'm good with either way, makes sense if many others will want to use it so we can move it later!